### PR TITLE
Feat: Adding basic feature for user management

### PR DESCRIPTION
**Pull request by ozendorf**
The goal is to add the simple possibility to allow explicitly users accessing to custom boards with a list of user emails in the config file.

### CODE REVIEW

**Comment by fjogeleit** [comment]
Thanks a lot for your effort, can you please re commit with the signed `-s` to fulfill the DCO.

File: backend/pkg/server/api/handler.go [hunk 1]
```diff
@@ -108,76 +111,95 @@ func (h *Handler) GetResourceDetails(ctx *gin.Context) {
...
-	}
+	session := sessions.Default(ctx)
 
-	endpoints, ok := h.clients[ctx.Param("cluster")]
-	if !ok {
-		ctx.AbortWithStatus(http.StatusNotFound)
-		return
-	}
+    // Get the profile from the session
+    profile := session.Get("profile")
```
Comment (fjogeleit) [inline backend/pkg/server/api/handler.go:117]: You can use `auth.ProfileFrom(ctx)` which returns `nil` if no profile is available

File: backend/pkg/server/api/handler.go [hunk 2]
```diff
@@ -108,76 +111,95 @@ func (h *Handler) GetResourceDetails(ctx *gin.Context) {
...
 
-	query := ctx.Request.URL.Query()
+	userProfile, ok := profile.(auth.Profile)
+    if !ok {
```
Comment (fjogeleit) [inline backend/pkg/server/api/handler.go:120]: also ensure that everything still works if authentication is not enabled

File: backend/pkg/server/api/handler.go [hunk 3]
```diff
@@ -108,76 +111,95 @@ func (h *Handler) GetResourceDetails(ctx *gin.Context) {
...
+    // Now you can access individual fields of the profile
+    email := userProfile.Email
+	config, ok := h.boards[ctx.Param("id")]
+	users := config.Users.List
 
-	if len(sources) == 0 {
-		sources, err = endpoints.Core.ListSources(ctx, url.Values{})
-		if err != nil {
-			ctx.AbortWithError(http.StatusInternalServerError, err)
+    if slices.Contains(users, email) {
```
Comment (fjogeleit) [inline backend/pkg/server/api/handler.go:130]: what happens when a custom board has an empty user list? In this case it should also be allowed.

Another suggestion would be to use a guard clause to make the code easier.

```suggestion
    if !slices.Contains(users, email) {
        ctx.AbortWithStatus(http.StatusUnauthorized)
        return
    }
```

File: backend/pkg/config/config.go [hunk 4]
```diff
@@ -220,6 +223,7 @@ type Config struct {
 	OAuth         OAuth          `mapstructure:"oauth"`
 	CustomBoards  []CustomBoard  `mapstructure:"customBoards"`
 	Local         bool           `mapstructure:"local"`
+	Users         string         `mastructure:"users"`
```
Comment (fjogeleit) [inline backend/pkg/config/config.go:226]: I would like to add an `Board` option around to give this users config a bit more context. Also its not used right now?

Comment (ozendorf) [reply backend/pkg/config/config.go]: yes you're right. I removed it.

**Pull request merged**